pkg/engine/executor/pipeline_test.go

```diff
@@ -0,0 +1,494 @@
+package executor
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/apache/arrow-go/v18/arrow"
+	"github.com/apache/arrow-go/v18/arrow/array"
+	"github.com/apache/arrow-go/v18/arrow/memory"
+	"github.com/stretchr/testify/require"
+)
+
+// CSVToArrow converts a CSV string to an Arrow record based on the provided schema.
+// It supports int64, float64, and boolean types for now.
+func CSVToArrow(fields []arrow.Field, csvData string) (arrow.Record, error) {
+	// Create schema
+	schema := arrow.NewSchema(fields, nil)
+
+	// Prepare memory allocator
+	mem := memory.NewGoAllocator()
+
+	// Create record builder
+	builder := array.NewRecordBuilder(mem, schema)
+	defer builder.Release()
+
+	// Parse CSV data
+	lines := strings.Split(strings.TrimSpace(csvData), "\n")
+	if len(lines) == 0 {
+		return nil, errors.New("empty CSV data")
+	}
+
+	// Process each line
+	for _, line := range lines {
+		if line == "" {
+			continue
+		}
+
+		// Split into values
+		values := strings.Split(line, ",")
```

Comment:
I'm not sure why we're not using Arrow's existing [csv package](https://pkg.go.dev/github.com/apache/arrow-go/v18@v18.2.0/arrow/csv) here?